File: compute-platform-core/src/main/java/org/lsw/core/operators/data/DataSource.java
package org.lsw.core.operators.data;

import org.lsw.core.model.ComputeData;

/**
 * 数据输入
 * @param <OUT>
 * @param <O>
 */
public abstract class DataSource<OUT, O extends DataSource<OUT, O>> extends ComputeData<OUT> {
    public DataSource(String name) {
        super(name);
    }

    /**
     * 读取数据
     * @return 读取到的数据
     */
    public abstract ComputeData<OUT> load();
}

File: compute-platform-spark/src/main/scala/org/lsw/compute/spark/api/data/DataSource.scala
package org.lsw.compute.spark.api.data

import org.apache.spark.sql.Dataset
import org.lsw.compute.spark.api.TransformBase

/**
 * 数据源抽象
 *
 * @tparam OUT 输出的数据类型
 */
abstract class DataSource[OUT] extends TransformBase[OUT] {


  /**
   * 得到Dataset[OUT]的实现。
   */
  override private[api] def transform(): Dataset[OUT] = {
    load()
  }

  /**
   * 读取数据的实现
   */
  protected def load(): Dataset[OUT]
}
